converter/dicom-cast/src/Microsoft.Health.DicomCast.Core/Features/Worker/FhirTransaction/Patient/PatientBirthDateSynchronizer.cs

```diff
@@ -0,0 +1,34 @@
+﻿// -------------------------------------------------------------------------------------------------
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License (MIT). See LICENSE in the repo root for license information.
+// -------------------------------------------------------------------------------------------------
+
+using Dicom;
+using EnsureThat;
+using Hl7.Fhir.Model;
+using Microsoft.Health.DicomCast.Core.Extensions;
+
+namespace Microsoft.Health.DicomCast.Core.Features.Worker.FhirTransaction
+{
+    /// <summary>
+    /// Provides functionality to synchronize DICOM properties to a specific <see cref="Patient.BirthDate"/> property.
+    /// </summary>
+    public class PatientBirthDateSynchronizer : IPatientPropertySynchronizer
+    {
+        /// <inheritdoc/>
+        public void Synchronize(DicomDataset dataset, Patient patient, bool newPatient)
+        {
+            if (newPatient)
+            {
+                EnsureArg.IsNotNull(dataset, nameof(dataset));
+                EnsureArg.IsNotNull(patient, nameof(patient));
+
+                FhirDateTime dicomPatientBirthDate = dataset.GetDatePropertyIfNotDefaultValue(DicomTag.PatientBirthDate);
+                if (dicomPatientBirthDate != null)
+                {
+                    patient.BirthDate = dicomPatientBirthDate.ToString();
```

Comment:
One thing you might want to consider is invalid date (for example, 2001/02/29). I am not sure if creating a new FhirDateTime would fail or not. If it does indeed fail, then this step will always fail, blocking the rest of the synchronization. Ideally the data is clean and birthdate is correctly but we have all seen bad data coming in. Just something to think about.

Reply:
The creation of the date does not fail per my tests! It will only fail if you call to assert whether or not it is a bad value.

Reply:
Added an assert in the extension so it returns null if the date is invalid in Fhir.